uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -909,39 +912,37 @@ public static function findOverlappingLeaveRequests($leaveRequestParams, $leaveR
     }
 
     $dateTimeSelector = $isCalculationUnitInHoursOrTOIL
-      ? "lr.from_date < '{$toDate}' AND lr.to_date > '{$fromDate}'"
-      : 'lrd.date BETWEEN %3 AND %4';
+      ? "lr.from_date < %7 AND lr.to_date > %6"
+      : "lrd.date BETWEEN %4 AND %5";
 
-    $absenceTypesIDs = array_merge($absenceTypesIDsWithSameUnit, $absenceTypesIDsWithDifferentUnit);
-    $absenceTypesSelector = $isCalculationUnitInHoursOrTOIL
-      ? (!$isTOIL
-        ? 'AND lr.type_id IN (' . implode(',', $absenceTypesIDsWithSameUnit) . ')'
-        : '')
-      : 'AND lr.type_id IN (' . implode(',', $absenceTypesIDs) . ')';
+    if ($isCalculationUnitInHours) {
+      $absenceTypesSelector = 'AND lr.type_id IN (' . implode(',', $absenceTypesIDsWithSameUnit) . ')';
 
-    $considerRequestsWithDifferentUnit = count($absenceTypesIDsWithDifferentUnit) && !$isTOIL
-      ? "OR ((lrd.date BETWEEN %3 AND %4) AND lr.type_id IN (" .
-        implode(',', $absenceTypesIDsWithDifferentUnit) . "))"
-      : "";
+      if (count($absenceTypesIDsWithDifferentUnit)) {
+        $absenceTypesSelector .=
+          "OR ((lrd.date BETWEEN %4 AND %5) AND lr.type_id IN (" .
+          implode(',', $absenceTypesIDsWithDifferentUnit) . "))";
+      }
+    } else if (!$isTOIL) {
+      $absenceTypesSelector = 'AND lr.type_id IN (' . implode(',', $absenceTypesIDs) . ')';
+    }
 
     $considerHalfDayRequests =
       ($fromDateType === $halfDayPMID
-        ? "AND (DATE_FORMAT(lr.to_date, '%y-%m-%d') != DATE_FORMAT('{$fromDate}', '%y-%m-%d') " .
+        ? "AND (DATE_FORMAT(lr.to_date, '%Y-%m-%d') != %4 " .
           "OR lr.to_date_type != {$halfDayAMID})"
         : "") .
       ($toDateType === $halfDayAMID
-        ? "AND (DATE_FORMAT(lr.from_date, '%y-%m-%d') != DATE_FORMAT('{$toDate}', '%y-%m-%d') " .
+        ? "AND (DATE_FORMAT(lr.from_date, '%Y-%m-%d') != %5 " .
           "OR lr.from_date_type != {$halfDayPMID})"
         : "");
 
     $overlapSelector =
       "($dateTimeSelector $absenceTypesSelector) " .
-      ($isCalculationUnitInHoursOrTOIL
-        ? $considerRequestsWithDifferentUnit
-        : $considerHalfDayRequests);
+      (!$isCalculationUnitInHoursOrTOIL ? $considerHalfDayRequests : '');
```

Comment:
I see that the `dateTimeSelector` variable is no longer used, and it still exists?

Reply:
At line 941 it is actually used. https://github.com/civicrm/civihr/pull/2575/commits/5404622efe233ef82908d5281870e65ca3c970ec#diff-875730b8f60e0ebe0f4cce61f223c6c8R941